core/src/main/scala/gnieh/diffson/JsonPointerSupport.scala

```diff
@@ -18,34 +18,67 @@ package gnieh.diffson
 
 import scala.annotation.tailrec
 
+import scala.language.implicitConversions
+
 trait JsonPointerSupport[JsValue] {
   this: DiffsonInstance[JsValue] =>
 
   import provider._
 
-  implicit val pointer = new JsonPointer()
-
   /** A class to work with Json pointers according to http://tools.ietf.org/html/rfc6901.
    *  The behavior in case of invalid pointer is customizable by passing an error handler
    *  when instantiating.
    *
    *  @author Lucas Satabin
    */
-  class JsonPointer {
+  case class JsonPointer(path: Pointer) {
+
+    /** Evaluates the given path in the given JSON object.
+     *  Upon missing elements in value, the error handler is called with the current value and element
+     */
+    final def evaluate(value: JsValue): JsValue = evaluate(value, this, Pointer.Root)
+
+    def /(key: String): JsonPointer = path :+ Left(key)
+
+    def /(idx: Int): JsonPointer = path :+ Right(idx)
+
+    def +:(part: Part): JsonPointer = copy(part +: path)
+
+    def :+(part: Part): JsonPointer = copy(path :+ part)
 
-    private var errorHandler: PointerErrorHandler = null
+    def serialize: String = if (path.isEmpty) "" else "/" + path.map {
+      case Left(l)  => l.replace("~", "~0").replace("/", "~1")
+      case Right(r) => r.toString
+    }.mkString("/")
 
-    private def handle(value: JsValue, pointer: String, parent: Pointer): JsValue =
-      if (errorHandler == null)
-        defaultHandler(value, pointer, parent)
-      else
-        errorHandler.orElse(defaultHandler)(value, pointer, parent)
+    @tailrec
+    private def evaluate(value: JsValue, path: JsonPointer, parent: JsonPointer): JsValue = (value, path.path) match {
```

Comment:
The error handler should be an implicit parameter of this method, allowing for overriding the default one at call site.